### PR TITLE
CBG-634: Fix intermittent failures in TLS configuration test

### PR DESCRIPTION
**Pull request by sarathkumarsivan**

### CODE REVIEW

File: base/bucket_test.go [hunk 1]
```diff
@@ -302,14 +305,23 @@ func TestGetViewQueryTimeout(t *testing.T) {
 	assert.Equal(t, expectedViewQueryTimeout, fakeBucketSpec.GetViewQueryTimeout())
 }
 
-var (
-	rootKeyPath    = filepath.Join(os.TempDir(), "root.key")
-	rootCertPath   = filepath.Join(os.TempDir(), "root.pem")
-	clientKeyPath  = filepath.Join(os.TempDir(), "client.key")
-	clientCertPath = filepath.Join(os.TempDir(), "client.pem")
-)
+func mockCertificatesAndKeys(t *testing.T) map[string]string {
```
Comment (bbrks) [inline base/bucket_test.go]: Returning a map of file paths here feels a little bit hard to follow. The consumer of this has to know the internal detail of this function (the map key names) in order to get those values back out.

Can we return a bunch of named strings instead?

```suggestion
func mockCertificatesAndKeys(t *testing.T) (certPath, clientCertPath, clientKeyPath, rootCertPath, rootKeyPath string) {
```

File: base/bucket_test.go [hunk 2]
```diff
@@ -358,44 +370,64 @@ func mockCertificatesAndKeys(t *testing.T) {
...
+	rootKeyPath := certPaths["rootKeyPath"]
+	rootCertPath := certPaths["rootCertPath"]
+	clientKeyPath := certPaths["clientKeyPath"]
+	clientCertPath := certPaths["clientCertPath"]
+	log.Printf("certPaths: %s", certPaths)
+
+	require.True(t, fileExists(rootKeyPath), "File %v should exists", rootKeyPath)
+	require.True(t, fileExists(rootCertPath), "File %v should exists", rootCertPath)
+	require.True(t, fileExists(clientKeyPath), "File %v should exists", clientKeyPath)
+	require.True(t, fileExists(clientCertPath), "File %v should exists", clientCertPath)
```
Comment (bbrks) [inline base/bucket_test.go]: `mockCertificatesAndKeys` already has some assertions in it, so we could verify the file paths exist in there, before the function returns anything.

File: base/bucket_test.go [hunk 3]
```diff
@@ -358,44 +370,64 @@ func mockCertificatesAndKeys(t *testing.T) {
 	certBytes, err = x509.CreateCertificate(rand.Reader, &clientTemplate, &rootTemplate, &clientKey.PublicKey, rootKey)
 	assert.NoError(t, err, "Client certificate should be generated")
```
Comment (bbrks) [inline base/bucket_test.go]: This and the errors above I think should be `require.NoError`. The tests won't work if this isn't successful.

**Review by bbrks — CHANGES_REQUESTED** [review]

File: base/bucket_test.go (hunk 1 above)
Comment (sarathkumarsivan) [reply base/bucket_test.go]: Makes sense. This has been fixed in latest revision.

File: base/bucket_test.go (hunk 3 above)
Comment (sarathkumarsivan) [reply base/bucket_test.go]: This is fixed in latest revision.

File: base/bucket_test.go (hunk 2 above)
Comment (sarathkumarsivan) [reply base/bucket_test.go]: Yeah, it would be better to verify the existence of file path inside mockCertificatesAndKeys; fixed in latest revision.

**Review by bbrks — CHANGES_REQUESTED** [review]
Minor tidy up - otherwise looks good.

File: base/bucket_test.go [hunk 4]
```diff
@@ -356,46 +361,62 @@ func mockCertificatesAndKeys(t *testing.T) {
...
-	assert.NoError(t, err, "Client certificate should be generated")
+	require.NoError(t, err, "Client certificate should be generated")
 	saveAsCertFile(t, clientCertPath, certBytes)
+
+	require.True(t, fileExists(rootKeyPath), "File %v should exists", rootKeyPath)
+	require.True(t, fileExists(rootCertPath), "File %v should exists", rootCertPath)
+	require.True(t, fileExists(clientKeyPath), "File %v should exists", clientKeyPath)
+	require.True(t, fileExists(clientCertPath), "File %v should exists", clientCertPath)
+
+	return
```
[the remark this answers is not in the document]
Comment (sarathkumarsivan) [reply base/bucket_test.go]: Makes more sense, naked return has been avoided in latest revision.

File: base/bucket_test.go [hunk 5]
```diff
@@ -356,46 +361,62 @@ func mockCertificatesAndKeys(t *testing.T) {
...
 	}
-	return nil
+	return info.IsDir()
 }
 
 func TestTLSConfig(t *testing.T) {
 	// Mock fake root CA and client certificates for verification
-	mockCertificatesAndKeys(t)
+	certPath, clientCertPath, clientKeyPath, rootCertPath, rootKeyPath := mockCertificatesAndKeys(t)
+	log.Printf("certPath: %v\nclientCertPath: %v\nclientKeyPath: %v\nrootCertPath: %v\nrootKeyPath: %v", certPath,
```
[the remark this answers is not in the document]
Comment (sarathkumarsivan) [reply base/bucket_test.go]: Yeah, this can be avoided since we are already asserting the paths in `mockCertificatesAndKeys`. Fixed in latest revision.

File: base/bucket_test.go [hunk 6]
```diff
@@ -356,46 +361,62 @@ func mockCertificatesAndKeys(t *testing.T) {
...
+		clientCertPath, clientKeyPath, rootCertPath, rootKeyPath)
```
[the remark this answers is not in the document]
Comment (sarathkumarsivan) [reply base/bucket_test.go]: Fixed in latest revision.

**Review by bbrks — APPROVED** [review]
Changes look good - I'll merge after the build finishes.
